src/Fmt_ast.ml

```diff
@@ -3838,10 +3863,11 @@ and fmt_structure_item c ~last:last_item ?ext {ctx; ast= si} =
   | Pstr_open open_descr -> fmt_open_description c open_descr
   | Pstr_primitive vd -> fmt_value_description c ctx vd
   | Pstr_recmodule bindings ->
-      hvbox 0
-        (list_fl bindings (fun ~first ~last binding ->
-             fmt_module_binding c ctx ~rec_flag:true ~first binding
-             $ fmt_if (not last) "@,@\n" ))
+      fmt_recmodule c ctx bindings
+        (module Ast.Module_binding_fields)
+        (fun c ctx ~rec_flag ~first b ->
```

Comment:
matching `b` with `Mod` here?

And the similar changes around for the other call site of `fmt_recmodule` above. Doing this would allow to remove `Module_fields_getter` and its implementations I think.

Reply:
The matching won't be enough to get the original type back, but I simplified the implementation as much as I could.